### PR TITLE
Visual and experience improvements to existing sub navigation flow

### PR DESCRIPTION
**Pull request by adamziel**
## Description

Implementation of https://github.com/WordPress/gutenberg/issues/22087

*Before*

*After*

## How has this been tested?
1. Open post editor
1. Add a navigation block with two levels of nested menus
1. Confirm the nested menus are only visible after clicking on them
1. Confirm everything looks good visually
1. Save the post
1. Open the post on the actual site (outside of the editor)
1. Confirm the menu works the same as before this PR and looks good

## Types of changes
Non-breaking changes

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by draganescu** [comment]
Hey @adamziel any idea why to me the menus are transparent?

transparent

Theme: Ambitious (blocks) and TwentyTwenty (classic)

**Comment by adamziel** [comment]
@draganescu interesting! That background color should come from the following rule:

```css
.is-style-light:not(.has-background) .wp-block-navigation__container {
	background-color: $light-style-sub-menu-background-color;
}
```
from `navigation/style.scss`.

Is it missing in your editor? Is the style variation not assigned maybe?

**Comment by draganescu** [comment]
@adamziel the problem is that is-style-light does not apply by default for me and in any case the default should not be transparent ... right?

**Comment by noisysocks** [comment]
I'm also seeing the issue @draganescu mentioned in both Twenty Twenty and Gutenberg Starter Theme.

Other than that, the code LGTM and it's a big improvement when I test locally! Nice job 👍

Pinging @karmatosed for any feedback she has.

**Comment by talldan** [comment]
There's a bug report for the transparent submenus - https://github.com/WordPress/gutenberg/issues/21449

**Comment by adamziel** [comment]
I believe the problem with the background is that [getBlocksWithDefaultStylesApplied](https://github.com/WordPress/gutenberg/blob/6e901d097aa2a1389368326d4315d47d5d790cbc/packages/block-editor/src/store/actions.js#L253) only considers preferredStyleVariations and not the default block style. I'll proceed with this PR as it is and spin a new one to address/discuss it more.

**Comment by adamziel** [comment]
I extracted the appender change into a separate PR (https://github.com/WordPress/gutenberg/pull/22165).

The failing test here is also broken on master.

**Comment by adamziel** [comment]
Let's move the discussion about the transparent background here: https://github.com/WordPress/gutenberg/pull/22167

**Comment by mapk** [comment]
This is looking good! I saw one area that can be improved visually (excluding the glaring lack of background 😄 )

The top level of the 2nd subnav doesn't align to the top of the first subnav. I see that's because it aligns to the top of the focused/selected area of the parent nav item. This makes sense, but still looks odd. 

Screen Shot 2020-05-07 at 10 00 42 AM

Can we align the selected state AND the child level to the top on both accounts? Like this:

Screen Shot 2020-05-07 at 10 15 38 AM

@karmatosed does this feel right to you? Or did you have some mockups closer to G2 that should be applied here?

**Comment by karmatosed** [comment]
@mapk in mocks there was a little gap, so I'd be keen to see that come back in. For example:

Frame 3

**Comment by karmatosed** [comment]
@adamziel I'm going to hold off another design review for now, once it's refined a bit I will give one again though. On the whole, this is shaping up really well and I am excited to see these iterations thank you.

**Review by talldan — APPROVED** [review]
Code looks good. 👍 

I also noticed the same thing about the menus not quite being lined up with the link:
Screenshot 2020-05-08 at 2 17 42 pm

Lets try to get this merged and improve that separately.

**Comment by adamziel** [comment]
I fixed the vertical alignment of sub-menus and merged this one. Let's do more improvements in another PR.

**Comment by adamziel** [comment]
See https://github.com/WordPress/gutenberg/pull/22228 and https://github.com/WordPress/gutenberg/pull/22227

**Comment by adamziel** [comment]
@karmatosed both PRs are now merged. What we did not reach a consensus on just yet, is whether or not the top-level navigation should have a background by default:

* https://github.com/WordPress/gutenberg/pull/22228#pullrequestreview-408862804
* https://github.com/WordPress/gutenberg/pull/22167#issuecomment-625342666

Let's get it sorted out here and we should be good to marked this issue as resolved.

**Comment by mapk** [comment]
> What we did not reach a consensus on just yet, is whether or not the top-level navigation should have a background by default:

@adamziel, no other text/link based blocks (ie. Heading, Paragraph, Media+Text, Quote, etc.) have a default background that I'm aware of. I think we're pretty safe following that pattern here. A setting that allows a background color to be added works well, but it's not necessary as a default.

**Comment by draganescu** [comment]
@mapk I wonder if any other blocks have z-index stacked parts? Because that is the issue with no default background, see through things appear broken.

**Comment by mapk** [comment]
@draganescu, when a user adds a Nav block to their Header (in the near future), they'll likely add the Nav block on top of a solid color Header. It is also likely they'll want the background color of the Header to show through the Navigation block. I foresee this being a highly common experience based on web patterns out there.

Now, any elements that are z-indexed on top of the Nav block (ie. submenus) should definitely have a background color. I'm totally in favor of this, because it feels like this is the only time where we experience a z-indexing issue of content on top of other content.

Here's a quick diagram to help communicate my direction.

background

The user can always add a background color to the block with the Color Settings.

**Comment by draganescu** [comment]
We should create an issue that explains this background setting behavior.

**Pull request merged**
